### ISSUE TITLE
Principled BSDF: Optimization of `sheen_tint.value` raises an Exception, if `sheen.value` is 0 (the default).

### ISSUE THREAD

**Issue by maxfrei750** [mitsuba-renderer/mitsuba3#339]
## Summary

Trying to optimize `sheen_tint.value` raises an Exception, if `sheen.value` is 0 (the default):

```
Exception has occurred: Exception
The contribution computed by the differential rendering phase is not attached to the AD graph! Raising an exception since this is usually indicative of a bug (for example, you may have forgotten to call dr.enable_grad(..) on one of the scene parameters, or you may be trying to optimize a parameter that does not generate derivatives in detached PRB.)
  File "/src/mitsuba3/build/python/mitsuba/python/ad/integrators/prb.py", line 229, in sample
    raise Exception(
  File "/src/mitsuba3/build/python/mitsuba/python/ad/integrators/common.py", line 947, in render_backward
    L_2, valid_2, state_out_2 = self.sample(
  File "/src/mitsuba3/build/python/mitsuba/python/util.py", line 400, in backward
    self.integrator.render_backward(self.scene, self.params, self.grad_out(),
  File "/src/mitsuba3/build/python/drjit/router.py", line 4346, in traverse
    dtype.traverse_(mode, flags)
  File "/src/mitsuba3/build/python/drjit/router.py", line 4465, in backward_from
    traverse(ta, _dr.ADMode.Backward, flags)
  File "/src/mitsuba3/build/python/drjit/router.py", line 4519, in backward
    backward_from(arg, flags)
  File "/workspace/development/mwe_principled_bsdf_optimization.py", line 60, in mwe_bsdf_optimization
    dr.backward(loss)
  File "/workspace/development/mwe_principled_bsdf_optimization.py", line 68, in <module>
    mwe_bsdf_optimization()
```
I ran into this while examining #333, but IMHO it is a separate issue. While this makes sense to me, a more meaningful message would be nice.

## System configuration
```
System information:

  CPU: 12th Gen Intel(R) Core(TM) i9-12900K
  GPU: NVIDIA GeForce RTX 3090
  Python: 3.8.14 (default, Sep 26 2022, 11:28:17) [GCC 9.4.0]
  NVidia driver: 520.56.06
  LLVM: 10.0.0

  Dr.Jit: 0.2.1
  Mitsuba: 3.0.1
     Is custom build? True
     Compiled with: Clang 10.0.0
     Variants:
        scalar_rgb
        cuda_ad_rgb
        llvm_ad_rgb
        cuda_ad_mono
        llvm_ad_mono
        llvm_ad_spectral
        cuda_ad_spectral
```

## Steps to reproduce

```python
import drjit as dr
import mitsuba as mi

mi.set_variant("llvm_ad_rgb")

SCENE_DICT = {
    "type": "scene",
    "sphere": {
        "type": "sphere",
        "bsdf": {
            "type": "principled",
            "sheen": 0.5    # if this is not specified, a TypeError is raised, when trying to optimize `sheen_tint.value`
        },
    },
    "fill_light": {
        "type": "constant",
        "radiance": 1.0,
    },
    "integrator": {
        "type": "prb"
    },
    "sensor": {
        "type":
            "orthographic",
        "to_world":
            mi.ScalarTransform4f.look_at(origin=[0, 0, 100],
                                         target=[0, 0, 0],
                                         up=[0, 1, 0]),
        "film": {
            "type": "hdrfilm",
            "pixel_format": "luminance" if "mono" in mi.variant() else "rgb",    # pylint: disable=not-callable
        }
    },
}


def mse(image, image_ref):
    return dr.mean(dr.sqr(image - image_ref))


def mwe_bsdf_optimization():

    scene = mi.load_dict(SCENE_DICT)
    image_reference = mi.render(scene)

    parameters = mi.traverse(scene)
    key = "sphere.bsdf.sheen_tint.value"
    parameters[key] = 0.2
    parameters.update()

    optimizer = mi.ad.Adam(lr=0.05)
    optimizer[key] = parameters[key]
    parameters.update(optimizer)

    for _ in range(3):
        image = mi.render(scene, parameters)
        loss = mse(image, image_reference)
        dr.backward(loss)
        optimizer.step()
        parameters.update(optimizer)

    print("Success!")


if __name__ == "__main__":
    mwe_bsdf_optimization()
```

Thanks for your help.

**Comment by Speierers** [comment]
Hi @maxfrei750 ,

This error makes sense to me as when `sheen` is inactive, the `sheen_tint` value won't be used at all in the BSDF. 

I don't think a more meaningful message would help in this case. For instance, one could enable gradient tracking on all parameters of the `principled` BSDF and optimize it starting at some initial state. This would generate a lot of warnings if we were trying to be a little more cautious with the error messages.
